tests/admin_changelist/tests.py

```diff
@@ -153,6 +153,35 @@ def get_list_select_related(self, request):
         cl = ia.get_changelist_instance(request)
         self.assertEqual(cl.queryset.query.select_related, {'player': {}, 'band': {}})
 
+    def test_many_search_terms(self):
+        parent = Parent.objects.create(name='Mary')
+        Child.objects.create(parent=parent, name='Danielle')
+        Child.objects.create(parent=parent, name='Daniel')
+
+        m = ParentAdmin(Parent, custom_site)
+        request = self.factory.get('/parent/', data={SEARCH_VAR: 'daniel ' * 80})
+        request.user = self.superuser
+
+        cl = m.get_changelist_instance(request)
+        # Before: django.db.utils.OperationalError: at most 64 tables in a join
+        self.assertEqual(cl.queryset.count(), 1)
+
+    def test_related_field_multiple_search_terms(self):
+        """
+        Searches over multi-valued relationships now return rows from
+        related models only when all searched fields match that row.
+        """
+        parent = Parent.objects.create(name='Mary')
+        Child.objects.create(parent=parent, name='Danielle', age=18)
+        Child.objects.create(parent=parent, name='Daniel', age=19)
+
+        m = ParentAdminTwoSearchFields(Parent, custom_site)
+        request = self.factory.get('/parent/', data={SEARCH_VAR: 'danielle 19'})
```

Comment:
This test involves two fields on the related model (name and age), versus the tests just added involving one field. Is it worth keeping this example, or should I remove?